db/constraints.c

```diff
@@ -863,6 +865,84 @@ int verify_del_constraints(struct ireq *iq, void *trans, int *errout)
             /* here, we need to retry to verify the constraint */
             /* sub 1 to go to current constraint again */
             continue;
+        } else if(del_null) {
+            int err = 0, idx = 0;
+            unsigned long long newgenid;
+            if (iq->debug) {
+                reqprintf(iq, "VERBKYCNSTRT SET NULL ON DELETE TBL %s RRN %d ", bct->tablename, rrn);
+            }
+
+            rc = bdb_lock_tablename_read(thedb->bdb_env, bct->tablename, trans);
```

Comment:
This isn't about this commit specifically, but in general - we should be getting these locks higher up.  Maybe when processing USEDB in osql code, also get locks for dependent tables if we do delete or update, since we'll definitely need to read them.